YoCode/FrontEndCheck.cs

```diff
@@ -5,60 +5,98 @@
 using OpenQA.Selenium.Chrome;
 using OpenQA.Selenium.Support.UI;
 using System.IO;
+using Microsoft.Win32;
 
 namespace YoCode
 {
     internal class FrontEndCheck
     {
-        private readonly IWebDriver browser;
+        private static IWebDriver browser;
         private readonly string port;
+        private const string CHROME = "Google Chrome";
+        private const string FIREFOX = "Firefox";
+
+        public static bool running;
 
         public FrontEndCheck(string applicantsWebPort, string[] keyWords)
         {
+            if (string.IsNullOrEmpty(applicantsWebPort))
+            {
+                FrontEndEvidence.SetFailed("Could not retrieve the port number. Another program might be using it.");
+            }
+            running = true;
             FrontEndEvidence.FeatureTitle = "New feature found in front-end implementation";
+            DriverService service;
 
-            try
+            RegistryKey key = Registry.LocalMachine.OpenSubKey(@"SOFTWARE\Clients\StartMenuInternet");
+            var browsers = key.GetSubKeyNames();
+            
+            if (browsers.Any(a=>a.Contains(FIREFOX)))
             {
                 try
                 {
-                    var foxService = FirefoxDriverService.CreateDefaultService(Directory.GetCurrentDirectory());
-                    foxService.HideCommandPromptWindow = true;
+                    service = FirefoxDriverService.CreateDefaultService(Directory.GetCurrentDirectory());
+                    service.HideCommandPromptWindow = true;
                     port = applicantsWebPort;
                     var options = new FirefoxOptions();
                     options.AddArgument("--headless");
-
-                    browser = new FirefoxDriver(foxService, options);
+                    browser = new FirefoxDriver((FirefoxDriverService)service, options);
                 }
-                catch (Exception)
+                catch { };
+            }
+            else if (browsers.Any(a=>a.Contains(CHROME)))
+            {
+                try
                 {
-                    var chromeService = ChromeDriverService.CreateDefaultService(Directory.GetCurrentDirectory());
-                    chromeService.HideCommandPromptWindow = true;
+                    service = ChromeDriverService.CreateDefaultService(Directory.GetCurrentDirectory());
```

Comment:
The first three lines of each of these branches are the same. That suggests they can be moved out of the if statement?